### ISSUE TITLE
How do I add users to NetOpeer2

### ISSUE THREAD

**Issue by tonynii** [CESNET/netopeer2#779]
I want to add users to NetOpeer2, but still failed, anyone can help me?

I try to add user on netconf-server item, but cannot use this user to login. below is data：
```
    <netconf-server xmlns="urn:ietf:params:xml:ns:yang:ietf-netconf-server">
    <listen>
      <endpoint>
        <name>default-ssh</name>
        <ssh>
          <ssh-server-parameters>
            <client-authentication>
              <supported-authentication-methods>
                <publickey/>
                <passsword/>
                <other>interactive</other>
              </supported-authentication-methods>
              <users>
              <user>
              <name>netconf</name>
              <password>$0$Tester123</password>
              <authorized-key>
              <name>foobar</name>
              <algorithm>rsa2048</algorithm>
              <key-data>**********</key-data>
</authorized-key>
              </user>
              </users>
            </client-authentication>
          </ssh-server-parameters>
        </ssh>
      </endpoint>
    </listen>
  </netconf-server>
```

**Comment by michalvasko** [comment]
Yes, currently you can actually log only with users that exist locally on the machine, so just create a `netconf` user and it should work. It will likely be changed in future and work the way you would expect meaning only whatever users you configure will be usable for logging into *netopeer2-server*.

**Comment by tonynii** [comment]
thanks! 
by the way, I want to use ssh private key to login system by user 'netconf'. so I try to modify this node. but is also not work.

Could you help me!

my client is ncclient(python library)

**Comment by michalvasko** [comment]
Right, authorized keys are also currently configured just like for standard `ssh(1)` meaning you must store them in `~/.ssh/authorized_keys` of the user.

**Comment by rakichinni** [comment]
The query was closed 2020. Just wanted to check, whether support for user management at netopeer2 available in the latest releases?

**Comment by michalvasko** [comment]
Yes, the latest release authenticates only users present in its YANG configuration.

**Comment by rakichinni** [comment]
Thanks for confirming, which means the older deployments authenticate users using authorized_keys (for publickey **supported-authentication-methods**) from the host is **completely replaced with the users configured through YANG.**

Hope there is no CMake option/flag to enable/disable the older way of authenticating users from host users.

(README.md is referencing keys from authorized_keys files on the host. If authentication can be done only with users configured through YANG then the info in README.md may be not correct.)

**Comment by michalvasko** [comment]
If a user is to be authenticated, it **must** be explicitly mentioned in the YANG configuration. The credentials, however, can be both **explicitly** configured or **read from the system**. Meaning you can configure a specific user to be authenticated using public keys stored in their `authorized_keys`. There is some [example configuration](https://github.com/CESNET/netopeer2/blob/master/example_configuration/ssh_listen.xml).
